### DISCUSSION TITLE
PAG 导出遮罩失效, 透明度不正确

### DISCUSSION THREAD

**Discussion by Xiaolianmao** [Tencent/libpag#1576] [Q&A] [answered]
**制作环境:**
macOS Ventura 13.4 | Adobe After Effects 23.4.0(Build 53) | PAGViewer 4.3.3.84 (beta)

**问题截图:**
![遮罩失效](https://github.com/Tencent/libpag/assets/35087840/8ee22428-07b7-4bf7-aef7-2e97027a1854)
![透明度不正确](https://github.com/Tencent/libpag/assets/35087840/4090ba39-6622-46bd-ad37-1e7816da5156)

**制作PAG文件:**
[制作文件.zip](https://github.com/Tencent/libpag/files/12117472/default.zip)

**备注:**
我记得我1月份还是2月份的时候就提过一次遮罩失效的问题...都半年过去了 还没修复呢...

**Comment by FusionXu** [comment]
看了一下这应该是一个新的问题，我们这边再定位一下

**Comment by FusionXu** [comment, marked as answer]
这个案例比较特殊，把AE工程另存为18x的低版本文件，用你现在用的ae（高版本）打开这个文件，再导出就正常了
可以先这样规避这个问题，我们再定位解决

**Reply by Xiaolianmao** [reply]
感谢🙏

**Comment by newstarchen** [comment]
一、分析：
此問題的原因是AE在2023新版本中的操作方式放生了變更：之前AE版本做遮罩（trackMatte）時只能指定模式而不能指定圖層，默認就是將上面緊鄰的圖層用作遮罩，而2023新版本可以指定某個圖層作為遮罩。本例中有幾個圖層引用了非緊鄰的圖層作為遮罩，故而引起兼容性問題。
二、解決：
這個問題後續我們將會兼容。
目前可以從素材上進行規避，提供兩個規避方法：
1、遮罩選擇緊鄰的上一層；如果有多個圖層同時引用同一個圖層做遮罩，可以將該圖層複製並分開放置。
2、將AE工程另存為低版本（比如18x）再打開。（此方法實際上是AE自動完成了上面方法1的操作）
